cmd/controller-manager/app/options/options.go

```diff
@@ -64,6 +67,16 @@ type KubeSphereControllerManagerOptions struct {
 	//      "kubesphere.io/creator=" means reconcile applications with this label key
 	//      "!kubesphere.io/creator" means exclude applications with this key
 	ApplicationSelector string
+
+	// ControllerSelectors is the list of controller selectors to enable or disable controller.
+	// '*' means "all enabled by default controllers"
+	// 'foo' means "enable 'foo'"
+	// '-foo' means "disable 'foo'"
+	// first item for a particular name wins.
+	// e.g. '-foo,foo' means "disable foo", 'foo,-foo' means "enable foo"
+	// * has the lowest priority.
+	// e.g. *,-foo, means "disable 'foo'"
+	ControllerSelectors []string
```

Comment:
I think ControllerGates is a better name.

Reply:
I see, I will rename it . Thanks.